1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [27]:
import numpy as np

In [28]:
from scipy.stats.stats import pearsonr

In [6]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp.sort()

In [7]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks.sort()

In [9]:
Mzp = zp.mean()
Mzp

101.4

In [13]:
Mks = ks.mean()
Mks

709.9

In [14]:
kszp = zp*ks

In [17]:
Mkszp = kszp.mean()
Mkszp

81339.7

In [16]:
Cov_1 =  Mkszp - (Mzp*Mks)
Cov_1

9355.839999999997

In [20]:
np.cov(zp, ks, ddof = 0)

array([[ 3494.64,  9355.84],
       [ 9355.84, 30468.89]])

In [22]:
Std_zp = np.std(zp)

In [23]:
Std_ks = np.std(ks)

In [24]:
Pearson_coef = Cov_1/(Std_zp*Std_ks)
Pearson_coef

0.9066783546151521

In [30]:
pearsonr(zp, ks)

(0.9066783546151527, 0.00029608989103534634)

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [31]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [32]:
x.sort()

In [33]:
sr1 = np.mean(x)
sr1

118.1

In [34]:
sr_std1 = np.std(x, ddof=1)
sr_std1

10.54566788359614

In [35]:
M = 118.1
std1 = 10.55
n1 = 10

In [36]:
t = 2.262

In [37]:
M-(t*std1/np.sqrt(n1))

110.55350896899758

In [38]:
M+(t*std1/np.sqrt(n1))

125.64649103100241

[110.55; 125.65]

3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [39]:
M2 = 174.2

In [40]:
std2 = 5

In [41]:
z= 1.96

In [42]:
n2 = 27

In [44]:
M2-(z*std2/np.sqrt(n2))

172.31398912064722

In [45]:
M2+(z*std2/np.sqrt(n2))

176.08601087935276

[172.31; 176.09]